packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv-remote-adapter.ts

```diff
@@ -0,0 +1,156 @@
+
+import type { Express } from "express";
+import { Optional } from "typescript-optional";
+
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+import {
+  ICactusPlugin,
+  ICactusPluginOptions,
+  IPluginWebService,
+  IWebServiceEndpoint,
+  PluginAspect,
+} from "@hyperledger/cactus-core-api";
+
+import { DefaultApi } from "./generated/openapi/typescript-axios";
+
+export interface IPluginKeychainAzureKvRemoteAdapterOptions
+  extends ICactusPluginOptions {
+  backend: DefaultApi;
+  keychainId: string;
+  logLevel?: LogLevelDesc;
+}
+
+/**
+ * Class responsible for ecapsulating an API client object and then acting as
+ * an adapter (ta-da) between said API client object and the calling code to
+ * which it is (should be) transparent whether it is talking to an in-process
+ * plugin instance of the keychain plugin or an adapter backed by an API client
+ * object in which scenario the real keychain plugin object can be anywhere
+ * else on the network and also can be written in any programming language that
+ * the author so desires.
+ */
+export class PluginKeychainAzureKvRemoteAdapter
+  implements ICactusPlugin, IPluginWebService {
+  public static readonly CLASS_NAME = "PluginKeychainAzureKvRemoteAdapter";
+
+  private readonly instanceId: string;
+  private readonly keychainId: string;
+  private readonly log: Logger;
+  private readonly backend: DefaultApi;
+
+  public get className(): string {
+    return PluginKeychainAzureKvRemoteAdapter.CLASS_NAME;
+  }
+
+  constructor(
+    public readonly opts: IPluginKeychainAzureKvRemoteAdapterOptions,
+  ) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.keychainId, `${fnTag} arg options.keychainId`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    Checks.nonBlankString(opts.keychainId, `${fnTag} options.keychainId`);
+    Checks.truthy(opts.backend, `${fnTag} options.backend`);
+    Checks.truthy(
+      opts.backend instanceof DefaultApi,
+      `${fnTag} opts.backend instanceof DefaultApi`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.opts.instanceId;
+    this.backend = opts.backend;
+    this.keychainId = opts.keychainId;
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  /**
+   * Dummy implementation that wires no web services on the host API server
+   * because there is no need. All the functionality is implemented somewhere
+   * else on a host that's accessible through the network to this object
+   * (because this class is a remote adapter not an actual plugin impl.).
+   *
+   * @param _expressApp
+   */
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    return [];
```

Comment:
@jagpreetsinghsasan If not already, please create an issue to implement the web services (I may have already said this, please disregard if the issues are already in existence)

Reply:
Created issue #1009 for writing the README.md
Created issue #1010 for web services implementation